From 6c91b2513d7ad1bd4774c8559c4ceb200a0a9d61 Mon Sep 17 00:00:00 2001
From: Nikias Bassen <nikias@gmx.li>
Date: Fri, 30 Oct 2009 02:17:08 +0000
Subject: build fix: afc-backend does not depend on libgudev

---
diff --git a/configure.ac b/configure.ac
index d15db88..2d032fa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -329,7 +329,7 @@ msg_afc=no
 AFC_LIBS=
 AFC_CFLAGS=
 
-if test "x$enable_afc" != "xno" -a "x$msg_gudev" = "xyes" ; then
+if test "x$enable_afc" != "xno" ; then
   PKG_CHECK_EXISTS(libiphone-1.0 >= 0.9.4, msg_afc=yes)
 
   if test "x$msg_afc" = "xyes"; then
--
cgit v0.8.3.1
